### ISSUE TITLE
Role mapping across trusted clusters does not work for `kubernetes_users` traits

### ISSUE THREAD

**Issue by travelton** [gravitational/teleport#4833]
### Description

**What happened**:
When mapping roles from a root cluster to the leaf, `kubernetes_users` trait from the root cluster are not mapped to the leaf cluster role. 

**What you expected to happen**:
`kubernetes_users` traits defined in a root cluster role should be mapped to the associated leaf cluster role. 

**How to reproduce it (as minimally and precisely as possible)**:

Root Cluster Role:
```
kind: role
version: v3
metadata:
  name: admin-root
spec:
  allow:
    kubernetes_groups: ["{{external.kubernetes_groups}}"]]
    kubernetes_users: ["{{external.kubernetes_users}}"]]
```

Leaf Cluster Role:
```
kind: role
version: v3
metadata:
  name: admin-leaf
spec:
  allow:
    kubernetes_groups: ["{{internal.kubernetes_groups}}"]]
    kubernetes_users: ["{{internal.kubernetes_users}}"]]
```

Leaf Cluster TC:
```
kind: trusted_cluster
version: v2
spec:
  role_map:
    - remote: "admin-root"
      local: ["admin-leaf"]
```
Mapped kubernetes_users is blank:
```
DEBU [AUTH]      Mapped roles [admin-root] of remote user "user@domain.com" to local roles [admin-leaf] and traits map[kubernetes_groups:[teleport-okta-admin] kubernetes_users:[] logins:[root]]. auth/permissions.go:142
```
### Environment
- Teleport version (use `teleport version`): 4.4.4 Enterprise
- Tsh version (use `tsh version`): 4.4.4 Enterprise
- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware): AWS

**Comment by awly** [comment]
@travelton do you have a `kubernetes_users` trait in SSO on the root cluster?
Can you `tsh login` into the root cluster and paste the output of `tsh status -d`?

**Comment by travelton** [comment]
> @travelton do you have a `kubernetes_users` trait in SSO on the root cluster?
> Can you `tsh login` into the root cluster and paste the output of `tsh status -d`?

Correct. I will PM you, as this is affecting a customer.

**Comment by webvictim** [comment]
Similar to https://github.com/gravitational/teleport/issues/3376

**Comment by travelton** [comment]
@travelton To test static values and determine if this is an external or internal issue.

**Comment by travelton** [comment]
@awly Confirmed, with a static user in the `kubernetes_users` on the root role, that user is not seen on the leaf cluster. 

```yaml
kind: role
metadata:
  name: admin
spec:
  allow:
    kubernetes_groups:
    - system:masters
    kubernetes_users:
    - travis
    logins:
    - '{{internal.logins}}'
    - root
```

```
DEBU [AUTH]      Mapped roles [admin local-admin] of remote user "travelton" to local roles [admin] and traits map[kubernetes_groups:[system:masters] kubernetes_users:[] logins:[travelton root]]. auth/permissions.go:143
```

I reviewed my x509 cert, after re-authenticating with my root cluster, but it actually has `kubernetes_user` set as null. Seems like the root cluster is not writing `kubernetes_users` into the cert? 

```
Subject: <omitted>  postalCode = "{\"kubernetes_groups\":[\"system:masters\"],\"kubernetes_users\":null,\"logins\":[\"travelton\"]}", O = admin + O = local-admin, <omitted>
```

**Comment by awly** [comment]
I can not reproduce this on our `master` branch.
It's possible this was fixed during the 5.0 release.

@travelton please check again with teleport 5.1.
If you don't have an environment to reproduce this, let's close the issue anyway since I tried following the above steps (using a local user).

**Comment by awly** [comment]
@travelton I'll close this for now, since there is no repro.
